contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +171,44 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
```

Comment:
2 reasons for this:

1. The "never expire" default is a bit on the sketchy side from a security point of view.  Having an option from the start is a small way to start mitigating it.
2. Even having this parameter there with a help string that says "forever" will clue people in to what the default behavior is.  This is a cheap way to document something that is somewhat security sensitive.